server/routes/update-goal/UpdateGoalController.ts

```diff
@@ -53,8 +51,28 @@ export default class UpdateGoalController {
     })
 
     await req.services.stepService.saveAllSteps(updated, uuid)
+  }
+
+  private async addNoteToGoal(req: Request, uuid: string, note: string) {
+    const goalData: Partial<NewGoal> = {
+      note,
+    }
 
-    return res.redirect(`${URLs.PLAN_OVERVIEW}`)
+    await req.services.goalService.updateGoal(goalData, uuid)
+  }
+
+  private saveAndRedirect = async (req: Request, res: Response, next: NextFunction) => {
+    const { uuid } = req.params
+    const { steps } = req.body
+    const note = req.body.moreDetail
+
+    try {
+      await this.updateSteps(req, uuid, steps)
+      await this.addNoteToGoal(req, uuid, note)
```

Comment:
Was thinking about this, as one may succeed and one may fail, as the steps may be updated but without the note.

Reply:
Resolved in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/180/commits/627a4b1fd89717c29d050b0f4e426e2b71b4cb2b